src/form/custom-fields.ts

```diff
@@ -45,3 +48,136 @@ export function createCustomFieldExample(): SerializedFormField {
     maxLength: 250
   }
 }
+
+type ArrayElement<ArrayType> = ArrayType extends readonly (infer ElementType)[]
+  ? ElementType
+  : never
+
+const idTypeOptions = [
+  {
+    value: 'NATIONAL_ID' as const,
+    label: {
+      defaultMessage: 'National ID',
+      description: 'Option for form field: Type of ID',
+      id: 'form.field.label.iDTypeNationalID'
+    }
+  },
+  {
+    value: 'PASSPORT' as const,
+    label: {
+      defaultMessage: 'Passport',
+      description: 'Option for form field: Type of ID',
+      id: 'form.field.label.iDTypePassport'
+    }
+  },
+  {
+    value: 'BIRTH_REGISTRATION_NUMBER' as const,
+    label: {
+      defaultMessage: 'Birth Registration Number',
+      description: 'Option for form field: Type of ID',
+      id: 'form.field.label.iDTypeBRN'
+    }
+  },
+  {
+    value: 'NONE' as const,
+    label: {
+      defaultMessage: 'None',
+      description: 'Option for form field: Type of ID',
+      id: 'form.field.label.iDTypeNone'
+    }
+  }
+]
+
+type IDType = ArrayElement<typeof idTypeOptions>['value']
+
+export function getIDType(
+  event: string,
+  sectionId: string,
+  conditionals: Conditional[] = [],
+  required: boolean
+): SerializedFormField {
+  const fieldName: string = `${sectionId}IdType`
+  const fieldId: string = `${event}.${sectionId}.${sectionId}-view-group.${fieldName}`
+  return {
+    name: fieldName,
+    customQuestionMappingId: fieldId,
+    custom: true,
+    required,
+    type: 'SELECT_WITH_OPTIONS',
+    label: {
+      id: 'form.field.label.iDType',
+      description: 'A form field that asks for the type of ID.',
+      defaultMessage: 'Type of ID'
+    },
+    initialValue: '',
+    validator: [],
+    mapping: getCustomFieldMapping(fieldId),
+    placeholder: formMessageDescriptors.formSelectPlaceholder,
+    conditionals,
+    options: idTypeOptions
+  }
+}
+
+function getValidators(configCase: string, idValue: IDType) {
+  if (idValue === 'NATIONAL_ID') {
+    return getNationalIDValidators(configCase)
+  }
+  return []
+}
+
+export function getIDNumber(
+  sectionId: string,
+  idValue: IDType,
+  conditionals: Conditional[] = [],
+  required: boolean
+): SerializedFormField {
+  const fieldName: string = `${sectionId}IdNumber${sentenceCase(idValue)}`
+  const validators = getValidators(sectionId, idValue)
+  // eslint-disable-next-line no-console
+  console.log('Custom field addded with handlebar: ', fieldName)
+  return {
+    name: fieldName,
+    required,
+    custom: true,
```

Comment:
Do we need this field to be a custom one? If not then as it's a default field that we support, we need to add it as a valid optional field here: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/config/src/handlers/forms/validation.ts#L228

Reply:
Thanks @Zangetsu101, I don't think it needs to be a custom one, but rather an optional field. I am on it